benchmark/benchmark.py

```
@@ -60,6 +62,64 @@ def write_metadata(metadata: Dict[str, Any]) -> None:
         log.error("Failed to write metadata", exc_info=True)
 
 
+def detect_result_folder() -> Tuple[str, str]:
+    """
+    Detect the result folder path and source path for benchmark results using Hydra's internal API.
+    """
+    local_path = HydraConfig.get().runtime.output_dir
+    path = Path(local_path)
+
+    if "multirun" in str(path):
+        parts = str(path).split("multirun")
+        date_time_path = parts[1].lstrip("/\\")
+        
+        path_parts = date_time_path.split("/")
+        if path_parts and path_parts[-1].isdigit():
+            # use parent to include multirun.yaml
+            local_path = str(Path(local_path).parent)
+            date_time_path = "/".join(path_parts[:-1])
+        
+        s3_path = os.path.join("multirun", date_time_path)
+    elif "outputs" in str(path):
```

Does benchmark.py ever create outputs dir, isn't the default set to multirun? It is probably better to throw an exception if the output dir is not as expected?

---

Yes it does, the benchmarks can be run as single direct experiments with the  `hydra.mode=RUN` flag. I added it incase someone prefers to run a quick experiment that way. eg

` uv run benchmark.py -- s3_bucket=tadiwaom-mountpoint-test region=eu-west-1 hydra.mode=RUN application_workers=4 benchmarks.fio.direct_io=true benchmarks.fio.size_gib=10 mountpoint.fuse_threads=8`